pkg/awsutils/awsutils.go

```diff
@@ -819,52 +821,98 @@ func (cache *EC2InstanceMetadataCache) createENI(useCustomCfg bool, sg []*string
 
 	log.Infof("Trying to allocate %d IP addresses on new ENI", needIPs)
 	log.Debugf("PD enabled - %t", cache.enablePrefixDelegation)
-	input := &ec2.CreateNetworkInterfaceInput{}
 
-	if cache.enablePrefixDelegation {
-		input = &ec2.CreateNetworkInterfaceInput{
-			Description:       aws.String(eniDescription),
-			Groups:            aws.StringSlice(cache.securityGroups.SortedList()),
-			SubnetId:          aws.String(cache.subnetID),
-			TagSpecifications: tagSpec,
-			Ipv4PrefixCount:   aws.Int64(int64(needIPs)),
+	describeSubnetInput := &ec2.DescribeSubnetsInput{
```

Comment:
We should skip the whole describe subnet block when we are using custom networking. Splitting this into its own function will make that easier